astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -24,211 +30,585 @@ class NDArithmeticMixin(object):
+        compare_wcs: `bool`
+            see :meth:`NDArithmeticMixin.arithmetic` parameter description.
 
-    def subtract(self, operand, propagate_uncertainties=True):
-        if propagate_uncertainties:
-            propagate_uncertainties = "propagate_subtract"
+        Returns
+        -------
+        result_wcs: any type
+            The WCS information of the first operand is copyied if it is set.
+            If it was not set the second operand will be tryed. If that one
+            had no WCS too ``None`` is returned.
+        """
+
+        # ok, not really arithmetics but we need to check which wcs makes sense
+        # for the result and this is an ideal place to compare the two WCS,
+        # too.
+
+        if self.wcs is None and operand.wcs is None:
+            # None has any wcs information so the result should have None, no
+            # need to compare the WCS here.
+            return None
+        elif self.wcs is None:
+            # Only the second operand had WCS information so take this and
+            # issue a warning if comparison of WCS is requested.
+            if compare_wcs:
+                log.info("Only the second operand had some WCS information.")
+            return deepcopy(operand.wcs)
+        elif operand.wcs is None:
+            # Same as last case but reversed
+            if compare_wcs:
+                log.info("Only the first operand had some WCS information.")
+            return deepcopy(self.wcs)
         else:
-            propagate_uncertainties = None
-        return self._arithmetic(
-            operand, propagate_uncertainties, "subtraction", np.subtract)
+            # Both have WCS information. If requested compare the WCS and
+            # issue a warning if they are not equal
+            # TODO: Currently (astropy 1.04) this is a bit tricky because no
+            # two astropy.wcs.WCS are really identical even if they are copied
+            # so there is no reason to raise an Exception here, but issue the
+            # warning nevertheless. Really unfortunate... I currently worked
+            # around it so the warning differs if both are astropy.wcs.WCS
+            # instances.
+            if compare_wcs and self.wcs != operand.wcs:
+                if isinstance(self.wcs, WCS) and isinstance(operand.wcs, WCS):
+                    log.info("Astropy.WCS does not allow equality checks.")
+                else:
+                    log.info("WCS is not equal.")
+            return deepcopy(self.wcs)
+
+    def _arithmetic_meta(self, operation, operand, meta_kwds_operate,
+                         meta_kwds_set):
+        """
+        Calculate the resulting meta.
 
-    if isinstance(_arithmetic.__doc__, string_types):
-        subtract.__doc__ = _arithmetic.__doc__.format(name="Subtract",
-                                                      operator="-")
+        There are two ways of altering the resulting meta, one is by
```

Comment:
I'm actually not sure what corner cases might be out there. Currently supported is:
- setting meta keywords to a fixed value and
- ``multiplying`` (or add, ...) meta keywords with the other meta keyword and
- ``multiplying`` (or add, ...) the meta keyword with the data (if it is a scalar) of the other operand. 
Are there other use-cases? These were all I could think of.